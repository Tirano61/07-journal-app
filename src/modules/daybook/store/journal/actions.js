
import journalApi from "@/api/journalApi";

// export const myAction = async({ commit }) =>{

// }




export const loadEntries = async({ commit }) =>{
    
    const { data } = await journalApi.get('/entries.json');
    
    const entries = []
    for(let id of Object.keys( data )){
        
        entries.push({
            id,
            ...data[id]
        })
    }

    commit('setEntries', entries )

}

export const upDateEntries = async({ commit }, entries) =>{

    const { date, text }  = entries;
    const dataSave = { date, text };

    await journalApi.put(`/entries/${entries.id}.json`, dataSave);

   
    //El spret sse hace que js no pase el objeto por referencia (puede pasar) {...entries}
    commit('upDateEntries', {...entries});
}

export const createEntries = async({ commit }, entries) =>  {

    const { date, text }  = entries;
    const dataSave = { date, text };
    const { data } = await journalApi.post(`/entries.json`, dataSave);
        
    dataSave.id = data.name;

    commit('addEntries', dataSave);

    return data.name;

}












