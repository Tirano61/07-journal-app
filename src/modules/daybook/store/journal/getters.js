

// export const myGetters = ( state ) =>{
    // return state.lo que nececite
// }




export const getEntriesByTerm = ( state ) => ( term = '' ) => {
    if ( term.length === 0 ) return state.entries;

    return state.entries.filter( ent => ent.text.toLowerCase().includes( term.toLocaleLowerCase() ) );
        

    
};

export const getEntriesById = ( state ) => ( id = '' ) =>{
   
    // console.log(state.entries)
    const entry = state.entries.find(  ent => ent.id === id);

    if(!entry) return; 

    return { ...entry }; //TODO: probar

    

};