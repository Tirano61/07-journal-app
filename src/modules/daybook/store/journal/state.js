

export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(),
            date: new Date().toDateString(),
            Text: 'Anim laboris occaecat sint labore pariatur voluptate nostrud dolor anim consequat nisi ea. Voluptate quis et id deserunt deserunt irure tempor. Ipsum esse reprehenderit fugiat incididunt commodo cupidatat incididunt occaecat quis aliquip exercitation aliqua magna ullamco.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            Text: 'Duis adipisicing anim minim elit ea pariatur ex culpa proident exercitation id magna aliqua. Voluptate pariatur proident voluptate ea tempor id do nostrud aliquip ullamco eu cillum commodo culpa. Laborum mollit id incididunt anim. Ut ut sunt esse officia exercitation proident laborum qui nostrud veniam elit quis. Adipisicing enim ut irure consequat cillum dolore do exercitation ullamco aute eiusmod.',
            picture: null,
        },
        {
            id: new Date().getTime() + 1000,
            date: new Date().toDateString(),
            Text: 'Mollit ullamco in pariatur reprehenderit pariatur. Officia officia mollit adipisicing sit irure id aliqua dolore mollit qui. Fugiat amet eu officia ea laborum laboris consequat eu commodo culpa. Mollit Lorem irure mollit aliqua magna incididunt nostrud aute pariatur adipisicing tempor Lorem non.',
            picture: null,
        },
    ]
});