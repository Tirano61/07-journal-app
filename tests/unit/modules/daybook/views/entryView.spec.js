
import { createStore } from 'vuex'
import journal from '../../../../../src/modules/daybook/store/journal';
import { shallowMount } from "@vue/test-utils";
import { journalState } from '../../../mock-data/test-journal-state';
import EntryView from '../../../../../src/modules/daybook/views/EntryView.vue'
import Swal from "sweetalert2";


const createVuexStore = ( inicialState ) => createStore({
    modules: {
        journal: {
            ...journal,
            state: {...inicialState}            
        }
    }
})

jest.mock('sweetalert2', () => ({
    fire: jest.fn(),
    showLoading: jest.fn(),
    close: jest.fn()
}))


describe('Pruebas en el EntryView', () => {
    
    const store = createVuexStore( journalState )
    
    store.dispatch = jest.fn()

    const mockRouter = {
        push: jest.fn()
    }

    let wrapper ;

    beforeEach( ()=>{

        jest.clearAllMocks()

        wrapper = shallowMount( EntryView, {
            props: {
                id: '-MjL_umRoXAYun_8Z-ie'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [ store ]
            },
           
        });
    })


    test('Debe sacar al usuario por que el id no existe', () => {

        const wrapper = shallowMount( EntryView, {
            props:{
                id: 'sdfsdfsdfsdf'
            },
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins: [store],
            },
        });
        
        expect( mockRouter.push).toHaveBeenCalledWith( {name: 'no-entry'} )

    })

    test('Debe mostrar la entarda correctamente', () => {
        
        expect(wrapper.html()).toMatchSnapshot()
        expect( mockRouter.push ).not.toHaveBeenCalled()

    })
    
    test('Debe borrar la entrada y salir', ( done ) => {
        
        Swal.fire.mockReturnValueOnce(  Promise.resolve({ isConfirmed: true}))
        
        wrapper.find('.btn-danger').trigger('click')

        expect( Swal.fire ).toHaveBeenCalledWith({
            title: 'Está seguro',
            text: 'Una vez borrado no se recuperará',
            showDenyButton: true,
            confirmButtonText: 'Si estoy seguro'
        })

       
        setTimeout(() =>{
            expect(store.dispatch).toHaveBeenCalledWith('journal/deleteEntry', '-MjL_umRoXAYun_8Z-ie')
            expect(mockRouter.push).toHaveBeenCalled()
            done()
        }, 1)


    })
    
    
})
