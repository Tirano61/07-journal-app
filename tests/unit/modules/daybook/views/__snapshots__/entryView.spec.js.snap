// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el EntryView Debe mostrar la entarda correctamente 1`] = `
<div class="entry-title d-flex justify-content-between p-2">
  <div><span class="text-success fs-3  fw-bold">11</span><span class="mx-1 fs-3 ">Septiembre</span><span class="mx-2 fs-4  fw-light">2021, Sábado</span></div>
  <div><input type="file" accept="image/png, image/jpeg" style="display: none;"><button class="btn btn-danger mx-2"> Borrar <i class="fa fa-trash-alt"></i></button><button class="btn btn-primary"> Subir Foto <i class="fa fa-up-load"></i></button></div>
</div>
<hr>
<div class="d-flex flex-column px-3 h-75"><textarea placeholder="¿Que sucedió hoy?"></textarea></div>
<fab-stub icon="fa-save"></fab-stub><img src="https://res.cloudinary.com/ddkavwbkd/image/upload/v1631393297/cvvidqyxglyib5tnl6tp.png" alt="elefante" class="img-thumbnail">
<!--v-if-->
`;
